cores/esp32/esp32-hal-i2c.c

```diff
@@ -1609,6 +1609,9 @@ i2c_err_t i2cFlush(i2c_t * i2c)
 }
 
 i2c_err_t i2cWrite(i2c_t * i2c, uint16_t address, uint8_t* buff, uint16_t size, bool sendStop, uint16_t timeOutMillis){
+    if((i2c==NULL)||((size>0)&&(buff==NULL))) { // need to have location to store requested data
```

Comment:
this can possibly be simplified to:
```
if(!i2c||(size && !buff))
```

NULL is defined as zero..

Reply:
@atanisoft yes it could be obfuscated down to your concise example.  I like to write code such that assumptions are minimized.  I have found that I make more mistakes when I try to rely on assumptions.   
The `!i2c` starts off wrong in my mind.  `!`(not) starts right off describing a boolean operation, then up comes a pointer? So, my mind says "if not a pointer" **What?** What's **NOT** a pointer? Oh!, this construct means "if the value of i2c is not zero" well why don't you just write it that way, a few extra characters in a source file don't mean squat when the hard drive is measured in terabytes.

Chuck.